pages/platform/public-cloud/using_openstack_tokens_for_multiple_calls/guide.en-gb.md

```diff
@@ -0,0 +1,164 @@
+```
+
+For Windows users, you can follow this guide to export your environment variables : 
+
+[Set OpenStack Environment Variables](https://docs.ovh.com/gb/en/public-cloud/set-openstack-environment-variables/)
+
+### Step 1: Download and source your openrc file
+
+Go on your Control Panel, Public Cloud tab, User & Roles, click on the three dots button, then download openrc file of your Openstack user wanted and specify the region you want to perform actions in
+
+Once downloaded, edit your openrc file and add this line :
+
+```bash
+OS_PASSWORD=<your_password>
+```
+
+Please adapt this line with your Openstack User password given at the user's creation step
+
+Then, source the file you previously downloaded
+
+```bash
+source openrc.sh
+```
+
+### Step 2: Issue an Openstack Token
+
+
+> [!primary]
+>
+An Openstack token has 24 hours of validity
+
+For a better reliability, you can issue a token every 8 hours (as an example) to avoid proceeding actions with an expired token
+
+Please prefer token creation instead of running directly the wanted action when you want to proceed long actions like snapshots, shelves, image creations, ...
+>
+
+Once you have sourced your openrc file, run this command to issue a token
+
+```bash
+openstack token issue
+```
+
+This command should show you a similar output : 
+
+```bash
++------------+----------------------------------------------------------------+
+| Field      | Value                                                          |
++------------+----------------------------------------------------------------+
+| expires    | 2023-04-06T08:33:15+0000                                       |
+| id         | gAAAAA[...]                                                    |
+| project_id | 8a7[...]                                                       |
+| user_id    | f99[...]                                                       |
++------------+----------------------------------------------------------------+
+```
+
+You can now export token previously issued : 
+```bash
+export OS_TOKEN = gAAAAA[...]
+```
+
+You can also export directly your token with this command : 
+
+```bash
+export OS_TOKEN=$(openstack token issue -f value -c id)
+```
+
+### Step 3: Remove useless variable
+
+In order to use your issued token to make actions with your user, you have to unset the "OS_USER_DOMAIN_NAME" variable
+
+To do so, run this command:
+
+```bash
+unset OS_USER_DOMAIN_NAME
+```
+
+### Step 4: Use token to run commands
+
+Now that you have your token, you can use classic Openstack calls to manage your infrastructure
+
+```bash
+openstack --os-auth-type token <command>
+```
+
+Example : 
+
+```bash
+openstack --os-auth-type token image list
+```
+
+### Step 5: Revoke Openstack token
+
+Once you performed all the actions you wanted, you can revoke used token to avoid it to be used for any other actions
+
+To do so :
+
+```bash
+openstack --os-auth-type token token revoke <token_id>
```

Comment:
token token?
:)

Reply:
Yup, I've tried without the --os-auth-type and I've this error : 
openstack token revoke $OS_TOKEN
__init__() got an unexpected keyword argument 'token'

By using the --os-auth-type token, it works perfectly

Reply:
Damn, I understand now!
Sorry for this

Reply:
I was able to get it working on openstack cli version 5.1.2 and 6.0.0, cool